integration-tests/utils/aws/cloudWatchGetLogs.ts

```diff
@@ -1,141 +1,118 @@
-          break
-        } else {
-          containsAllPatterns = true
-        }
-      }
-      return containsAllPatterns
-    })
-    if (matchingEvent.length == 1) {
-      console.log(`Event found in log stream: ${latestLogStreamName}`)
-      eventMessage = matchingEvent[0].message as string
-      console.log(`MATCHED EVENT: ${eventMessage}`)
-      eventMatched = true
-      break
-    } else {
-      latestLogStreamName = await getLatestLogStreamName()
-    }
-  }
-  const result: EventLogStream = {
-    logStreamName: latestLogStreamName,
-    eventMessage: eventMessage
+const findMatchingLogEvents = async (
+  logGroupName: string,
+  logStreams: LogStream[],
+  filterPatterns: string[]
+) => {
+  const input = {
+    logGroupName: logGroupName,
+    logStreamNames: logStreams.map(
+      (logStream) => logStream?.logStreamName
+    ) as string[],
+    filterPattern: filterPatterns.map((pattern) => `"${pattern}"`).join(' ')
   }
+  const command = new FilterLogEventsCommand(input)
+  const response = await cloudWatchLogsClient.send(command)
 
-  return result
+  return response.events ?? []
 }
 
-const pause = (delay: number): Promise<unknown> => {
-  return new Promise((r) => setTimeout(r, delay))
-}
+const extractRequestIdFromEventMessage = (message: string) => {
+  const requestId = message.split('\t')[1]
+  console.log(`RequestId: ${requestId}`)
 
-export const getMatchingLogEvents = async (
-  filterPattern: string,
-  streamName: string
-): Promise<FilteredLogEvent[]> => {
-  const filterLogEventsParams: FilterLogEventsCommandInput = {
-    logGroupName: INITIATE_DATA_REQUEST_LAMBDA_LOG_GROUP,
-    logStreamNames: [streamName],
-    filterPattern: `${filterPattern}`
-  }
+  return requestId
+}
 
-  const filterLogEventsCommand = new FilterLogEventsCommand(
-    filterLogEventsParams
-  )
+const waitForEventWithPatterns = async (
+  logGroupName: string,
+  eventMessagePatterns: string[],
+  maxAttempts: number
+) => {
+  let attempts = 0
+  let logStreams = await getLogStreams(logGroupName)
+  let eventMatched = false
+  let logEvents: FilteredLogEvent[]
+
+  while (!eventMatched && attempts < maxAttempts) {
+    attempts++
+    logEvents = await findMatchingLogEvents(
+      logGroupName,
+      logStreams,
+      eventMessagePatterns
+    )
 
-  await pause(500)
-  const filterLogEventsResponse: FilterLogEventsCommandOutput =
-    await cloudWatchLogsClient.send(filterLogEventsCommand)
+    if (logEvents.length == 0) {
+      await pause(1000)
+      logStreams = await getLogStreams(logGroupName)
+      continue
+    }
 
-  const filterLogEvents: FilteredLogEvent[] =
-    filterLogEventsResponse.events ?? []
+    if (logEvents.length > 1) {
+      throw Error('More than 1 event matched, check filter patterns')
+    }
 
-  console.log(`FILTERED LOG EVENTS: ${filterLogEvents.length}`)
+    const message = logEvents[0]?.message as string
+    console.log(`Found event: ${message}`)
+    eventMatched = true
```

Comment:
I don't think you need this variable `eventMatched` now, because you're returning underneath, and the attempts < maxAttempts condition above is what stops the loop.

Reply:
Done